### ISSUE TITLE
v2 access from Objective-C

### ISSUE THREAD

**Issue by tjdhome** [dropbox/SwiftyDropbox#23]
I know that this code is intended for Swift code, but... I'm working on a TVos app and am have ported a boatload of Objective-C code to it. I'm adding in access to Dropbox folders and have added in a AppleTV activation flow using SwiftyDropbox on TVos (since the V1 ios Dropbox code seems abandoned and auth1 is such a pain).

So, here's my request: can you derive your top-level classes from NSObject? This will make the bridging code generation at least understand some things. The response objects are still problematic- since they are derived from CustomStringConvertible which is not accessible from obj-c. But I've added extensions to handle that.

**Comment by smarx** [comment]
We plan to enable access from Objective-C, but it's more complicated than deriving from `NSObject` and adding the `@objc` decorator. Notably, we use non-numeric enums all over. I believe we'll need to expose a substantial compatibility layer. Our plan is to address Objective-C access after the Swift SDK comes out of beta.

**Comment by tjdhome** [comment]
Terrific news that you're planning an Objective-C interface- I certainly won't complain! In the meantime I just forked the repo and made a few changes so that my particular use cases would work with minimal change to all sides. Just changing the derivation on a couple of your Swift objects made it really easy to create a bridging layer.

**Comment by varenc** [comment]
hey @tjdhome I'd be curious to see what you changed in our SDK to create a bridging layer.  When I looked at this earlier it seemed like it would be more work than that.

**Comment by tjdhome** [comment]
I added NSObject as the base class for the auth stuff and the Dropbox client. I also modified the access to the UIViewController so it would return the view controller that I wanted to manage. This gave me enough to be able to write a few thin Swift-Objective-C wrappers which got me going. I should note that on both our products that use Dropbox, I have evolved to only use SwiftyDropbox for auth and some file uploads. I found that it didn't handle too many use cases that I had - full control and monitoring over async uploads and downloads- access to some API V1 functions that don't seem to be supported in V2., yada yada yada. My programming model for all async operations is PromiseKit-based on both TVos and IOS and at the end of the day, it was easier to 'roll my own' file/NSData/ALAsset uploaders and downloaders. Also note- my code is almost entirely Objective-C- others mileage probably varies in Swift.
